tests/operators/istream_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+#include <bigint.hpp>
+
+int main()
+{
+    std::ifstream f("input.txt");
+    std::string line;
+
+    if (f.is_open())
```

Comment:
a `try` `catch` will be good in place of this `if` `else` isn't?

Reply:
But neither ifstream constructor nor is_open() raises an exception, right?

Reply:
http://www.cplusplus.com/reference/ios/ios/exceptions/

Reply:
Oh, got it. I'll add a try-catch👍🏻

Reply:
Okay, I read that documentation again and it says,

> "All streams have goodbit by default (they do not throw exceptions due to error state flags being set)."

So I am bit confused, do I still need to make these changes? Am I missing something?

Reply:
documentation text is a bit confusing, example mentioned below is a bit more clear
```C++
// ios::exceptions
#include <iostream>     // std::cerr
#include <fstream>      // std::ifstream

int main () {
  std::ifstream file;
  file.exceptions ( std::ifstream::failbit | std::ifstream::badbit );
  try {
    file.open ("test.txt");
    while (!file.eof()) file.get();
    file.close();
  }
  catch (std::ifstream::failure e) {
    std::cerr << "Exception opening/reading/closing file\n";
  }

  return 0;
}
```
it will throw an exception whenever ther is a failbit or badbit.
there is no harm in trying stuff. let's try this see what happens if there is some problem we will keep the if else for now.

Reply:
For that to happen someone will have to explicitly set `file.exceptions ( std::ifstream::failbit | std::ifstream::badbit );` as shown in the example. But we are not doing that. So do you want me to set failbit and badbit for ifstream object?

Reply:
leave this as it as, let's focus on other stuff, i am merging this for now.